go.mod

```diff
@@ -4,86 +4,84 @@ go 1.15
 
 require (
 	github.com/akutz/gofsutil v0.1.2
-	github.com/container-storage-interface/spec v1.2.0
+	github.com/container-storage-interface/spec v1.3.0
 	github.com/coreos/etcd v3.3.25+incompatible // indirect
 	github.com/davecgh/go-spew v1.1.1
 	github.com/docker/spdystream v0.0.0-20181023171402-6480d4af844c // indirect
 	github.com/elazarl/goproxy v0.0.0-20200710112657-153946a5f232 // indirect
 	github.com/elazarl/goproxy/ext v0.0.0-20200710112657-153946a5f232 // indirect
 	github.com/fsnotify/fsnotify v1.4.9
-	github.com/golang/protobuf v1.5.1
+	github.com/go-kit/kit v0.10.0 // indirect
+	github.com/golang/protobuf v1.5.2
 	github.com/google/uuid v1.2.0
-	github.com/googleapis/gnostic v0.5.4 // indirect
 	github.com/grpc-ecosystem/go-grpc-middleware v1.2.0 // indirect
 	github.com/grpc-ecosystem/grpc-gateway v1.14.5 // indirect
 	github.com/kr/text v0.2.0 // indirect
 	github.com/kubernetes-csi/csi-lib-utils v0.7.0
 	github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e // indirect
-	github.com/onsi/ginkgo v1.16.2
-	github.com/onsi/gomega v1.10.5
+	github.com/onsi/ginkgo v1.16.4
```

Comment:
run-e2e-test uses ginkgo v1.16.2 - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/hack/run-e2e-test.sh#L23

if this is not required for k8s v1.21, please revert it back. otherwise, change this in the script as well and make sure e2e tests are not affected due to ginkgo version bump up.

Reply:
ginkgo & gomega dependencies are updated by controller-runtime v0.9.0 here.
https://github.com/kubernetes-sigs/controller-runtime/blob/v0.9.0/go.mod#L13-L14

And we needed to update controller-runtime so that it's compatible with client-go v0.21.1.
Changed it in `run-e2e-test.sh`.